client/components/FaveButton.tsx

```diff
@@ -19,15 +19,40 @@ const FaveButton = ({ zoop }: FaveButtonProps) => {
   const currentUser = useSelector((state: RootState) => state.auth.user)
 
   const [addFave, {isLoading, data, error}] = useAddFaveMutation();
+  const [
+    removeFave, 
+    {
+      isLoading: isRemoveFaveLoading, 
+      data: removeFaveData, 
+      error: removeFaveError
+    }] = useRemoveFaveMutation();
 
 
+  const currentUserFave = zoop.faves.filter(
+    fave => fave.faverId === currentUser?.id
+  )
```

Comment:
I'd recommend using something like `find()` (or `some()`) here rather than filtering and checking for length

Reply:
Yeah, I wondered if find() would be better. I suppose truthy checks are better practice in this situation?